File: bitcoin.pas
function solve(N, E: longint; var C: array of longint): longint;
begin
    (* insert your code here *)
    solve := 42;
end;


const
    MAXN = 100000;
var
    N, E, i  : longint;
    C        : array[0..MAXN-1] of longint;
    fr, fw   : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, E);
    for i:=0 to N-1 do
        read(fr, C[i]);

    writeln(fw, solve(N, E, C));
    close(fr);
    close(fw);
end.

File: bitcoin.cpp
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int C[MAXN];

double solve(int N, double E, int C[]) {
    // insert your code here
    return 42;
}

int main() {
    FILE *fr, *fw;
    int N, E, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &E));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &C[i]));

    fprintf(fw, "%.8f\n", solve(N, E, C));
    fclose(fr);
    fclose(fw);
    return 0;
}